sdk/formrecognizer/ai-form-recognizer/src/transforms.ts

```diff
@@ -341,57 +364,127 @@ export function toRecognizeContentResultResponse(
   }
 }
 
-export function toFormModelResponse(response: GetCustomModelResponse): FormModelResponse {
-  const common = {
-    ...response.modelInfo,
-    trainingDocuments: response.trainResult?.trainingDocuments.map((d) => {
-      if (!Array.isArray(d.errors)) {
-        d.errors = [];
-      }
-      return d;
-    }),
-    errors: response.trainResult?.errors,
-    _response: response._response
-  };
+function flattenTrainingDocuments(
+  original: GetCustomModelResponse
+): TrainingDocumentInfo[] | undefined {
+  if (original.composedTrainResults) {
+    // Composed model, need to zip the training documents into a flat array and add modelID correlation.
+    const mappedResultDocuments = original.composedTrainResults.map((innerResult) =>
+      innerResult.trainingDocuments.map((info) => ({
+        ...info,
+        modelId: innerResult.modelId,
+        errors: info.errors ?? []
+      }))
+    );
 
-  if (response.modelInfo.status !== "ready") {
-    return common;
+    // Fold these together using a simple reduce
+    return mappedResultDocuments.reduce(
+      (prev, next) => Array.prototype.concat.call(prev, next),
+      []
+    );
+  } else if (original.trainResult) {
+    // Normal training scenario with only one trainResult
+    return original.trainResult.trainingDocuments.map((info) => ({
+      ...info,
+      modelId: original.modelInfo.modelId,
+      errors: info.errors ?? []
+    }));
   }
 
-  if (response.trainResult?.averageModelAccuracy || response.trainResult?.fields) {
-    // training with forms and labels, populate from trainingResult.fields
-    const fields: Record<string, CustomFormModelField> = {};
-    for (const f of response.trainResult.fields!) {
-      fields[f.fieldName] = { name: f.fieldName, accuracy: f.accuracy, label: null };
+  return undefined;
+}
+
+function toSubmodelsFromComposedTrainResults(results: TrainResult[]): CustomFormSubmodel[] {
+  const mappedSubmodels = results.map((r) => toSubmodelsFromTrainResultLabeled(r));
+
+  // Flatten the array
+  return mappedSubmodels.reduce((prev, next) => Array.prototype.concat.call(prev, next), []);
```

Comment:
why not just
```suggestion
  return mappedSubmodels.reduce((prev, next) => prev.concat(next), []);
```

Reply:
Oh yeah that would be simpler. Or even better `[].concat(...mappedSubmodels)`.

I had originally thought I was being clever and written this as just `reduce(Array.prototype.concat.call, [])`, but that doesn't work because of the extra arguments that are passed by `reduce`, and then I didn't think too hard about changing it.